Warning message for inactive token

jerabekjiri opened the pull request:
Issue: [AAH-727](https://issues.redhat.com/browse/AAH-727)

Added a warning message after the 'Load token' button is clicked. 
Note: the warning message _The token will expire after 30 days of inactivity. To renew the token, run the command below._ will be most likely be changed when I get the right text from Chris. 

Can you please check if this is the correct design approach with the code block example inside the alert?

after token is loaded:
![warning_message](https://user-images.githubusercontent.com/19647757/125481019-280cee08-491f-417f-989e-8cf6ea213a5d.png)

expanded codeblock:
![warning_message_expanded](https://user-images.githubusercontent.com/19647757/125477766-44f69ced-3d6c-494d-8c49-0eb99f356410.png)

jerabekjiri commented:
Just noticed, in the token renew command
`curl https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token -d grant_type=refresh_token -d client_id="{{ user_name }}" -d refresh_token="{{ user_token }}" --fail --silent --show-error --output /dev/null` 
should `{{ user_token }}` and `{{ user_token }}` be changed for real values?

ZitaNemeckova commented:
@sbuenafe-rh can you please have a look, thanks :)

@jerabekjiri I would say depends because you may not have the token generated when you read the message. So maybe just the `user_name`. WDYT @newswangerd ?

jerabekjiri commented:
> @jerabekjiri I would say depends because you may not have the token generated when you read the message. So maybe just the `user_name`.

@ZitaNemeckova Well the warning message is displayed after the primary 'Load token' button is clicked, so the token is always generated? Or should we just show the message whenever a user visits the API token page?

ZitaNemeckova commented:
@jerabekjiri You are right! I missed that it's displayed after clicking the button. In that case it makes sense to populate both with real value.

ZitaNemeckova commented:
@jerabekjiri please add a screenshot of whole page, please.

jerabekjiri commented:
@ZitaNemeckova Here you go
![warning_message_page](https://user-images.githubusercontent.com/19647757/125771509-1a8e8116-025a-4e68-adb5-df2a22cee0c1.png)

I was looking into that test failure, not really sure what to do here.

ZitaNemeckova commented:
@sbuenafe-rh Can you please have a look at the added warning at https://github.com/ansible/ansible-hub-ui/pull/635#issuecomment-880578071 from UX perspective. Thank you :)

jerabekjiri commented:
According to Chris Budzilowicz, the text in UI is good for now.

ZitaNemeckova commented:
Susan give it 👍 too.

himdel reviewed (commented):
Let's wait for Susan's feedback, I'd put the message *under* the API token section, closer to where the token is shown.

sbuenafe-rh commented:
@ZitaNemeckova @jerabekjiri After re-reading this PR and speaking with @himdel I understand that we cannot track the token, therefore we should not be using the inline alert. The inline alert causes me to think that the user needs to take immediate action. I would place the token inactivity expiration message in the API token card below the copy to clipboard component. :)

jerabekjiri commented:
- Moved the message below the token copy to clipboard component. 
- Changed it to an all-time visible message and once the user loads the token, the placeholder value (user_token) in the command is updated to the real token value. 

![token_updated](https://user-images.githubusercontent.com/19647757/126083900-5a145431-17b0-4536-a73b-804dbaeb4135.png)

ZitaNemeckova commented on src/containers/token/token-standalone.tsx:
```diff
@@ -1,10 +1,17 @@
+  ClipboardCopy,
+  ClipboardCopyVariant,
+  Button,
+  Alert,
```
`Alert` and `AlertActionCloseButton ` aren't used. Please remove them.

jerabekjiri replied:
Removed :)

himdel commented:
We also need to change the url to use the actual current URL.
Most instances won't be running on sso.redhat.com

newswangerd reviewed (changes requested):
The renew token message only applies to insights mode, not standalone mode. These changes should be put on `token-insights.tsx`.

The changes look good tho

jerabekjiri commented:
Moved the warning message to the insights mode and removed it in standalone. 
![Screenshot from 2021-07-20 17-54-56](https://user-images.githubusercontent.com/19647757/126381529-0a71b805-f119-4f20-92aa-a50d4185a23b.png)

jerabekjiri commented:
@newswangerd @himdel @ZitaNemeckova check changes pls :)

newswangerd commented:
Waiting on confirmation from the content team about how the renew script functions. Will update my review soon.

newswangerd commented on src/containers/token/token-insights.tsx:
```diff
@@ -82,6 +89,22 @@ class TokenPage extends React.Component<RouteComponentProps, IState> {
+              style={{ paddingTop: 'var(--pf-global--spacer--md)' }}
+            >
+              <span>
+                The token will expire after 30 days of inactivity. To renew the
```
So, this script only works before the token expires. Lets change the wording to something like "Run the command below periodically to prevent your token from expiring."

cbudz commented:
@newswangerd Yes, I think that language is fine.

newswangerd commented:
@jerabekjiri Oh, I didn't mean for you to get rid of the first part. The full text should be "The token will expire after 30 days of inactivity. Run the command below periodically to prevent your token from expiring."

Sorry for the confusion

jerabekjiri commented:
@newswangerd My fault, I didn't even think about it and changed it in hurry :facepalm: Should be fixed now :)

newswangerd reviewed (approved):
Looks good!

himdel reviewed (approved):
Works for me, LGTM :)

himdel commented:
(when doing a manual backport, also add #711 to make prettier happy :))
